File: searchimg.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys

if __name__ == '__main__':
    # Chrome WebDriver 로 chrome.exe 실행
    browser = webdriver.Chrome('./chromedriver.exe')

    # 사이트 호출
    browser.implicitly_wait(10)
    browser.get("https://www.google.co.kr/imghp?hl=ko&tab=wi&ogbl")

    # 검색창에 검색어 입력후 엔터
    serachstring = browser.find_element_by_name("q")
    serachstring.send_keys('이미지 캐릭터')
    serachstring.send_keys(Keys.RETURN)

    # 링크 클릭
    browser.find_element_by_link_text('애니메이션').click()

    # 페이지 읽기
    print(browser.page_source)

    # 드라이브 종료
    browser.quit()

File: golang/searchimg.go
// searchimg
package main

import (
	"fmt"
	"time"

	"github.com/tebeka/selenium"
)

const (
	path = `E:\GoApp\src\chromedriver.exe`
	port = 8080
)

func main() {

	opts := []selenium.ServiceOption{}

	server, err := selenium.NewChromeDriverService(path, port, opts...)
	if err != nil {
		fmt.Println(err)
	}
	defer server.Stop()

	c := selenium.Capabilities{"browserName": "chrome"}

	wd, err := selenium.NewRemote(c, fmt.Sprintf("http://localhost:%d/wd/hub", port))
	if err != nil {
		fmt.Println(err)
	}
	defer wd.Quit()

	//사이트 호출
	err = wd.Get("https://www.google.co.kr/imghp?hl=ko&tab=wi&ogbl")
	if err != nil {
		panic(err)
	}

	wd.SetImplicitWaitTimeout(10 * time.Second)

	//검색창에 검색어 입력후 엔터
	id, _ := wd.FindElement(selenium.ByName, `q`)
	id.Clear()
	id.SendKeys("이미지 캐릭터")
	id.SendKeys(selenium.EnterKey)

	wd.SetImplicitWaitTimeout(10 * time.Second)

	//링크 클릭
	link, _ := wd.FindElement(selenium.ByLinkText, "애니메이션")
	link.Click()

	wd.SetImplicitWaitTimeout(10 * time.Second)

	url, _ := wd.CurrentURL()
	title, _ := wd.Title()
	pagesource, _ := wd.PageSource()

	fmt.Println(url)
	fmt.Println(title)
	fmt.Println(pagesource)

	time.Sleep(60 * time.Second)
}
